### PR TITLE
[ember] refine types for Ember.tryInvoke

### PR DESCRIPTION
**Pull request by mike-north**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.emberjs.com/api/ember/3.4/functions/@ember%2Futils/tryInvoke
- * ~Increase the version number in the header if appropriate.~
- * ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

Resolves https://github.com/typed-ember/ember-cli-typescript/issues/257

### CODE REVIEW

File: types/ember/index.d.ts [hunk 1]
```diff
@@ -27,6 +27,20 @@ declare module 'ember' {
...
+            ? [A]
+            : F extends (a: infer A, b: infer B) => any
+                ? [A, B]
+                : F extends (a: infer A, b: infer B, c: infer C) => any
+                    ? [A, B, C]
+                    : F extends (a: infer A, b: infer B, c: infer C, d: infer D) => any
+                        ? [A, B, C, D]
+                        : F extends (a: infer A, b: infer B, c: infer C, d: infer D, e: infer E) => any
+                            ? [A, B, C, D, E]
+                            : never;
```
Comment (mike-north) [inline types/ember/index.d.ts]: I think we can all agree, this is pretty awful. Unfortunately, we can't use [inference of rest parameters](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#generic-rest-parameters) as long as we want to continue supporting TS 2.8 and 2.9

**Comment by mike-north** [comment]
Please wait on merging this. Typing the correct behavior is a little tricker than was initially apparent

**Comment by mike-north** [comment]
This is ready for review now. Test cases convince me that these typings are accurate

**Review by jamescdavis — APPROVED** [review]
FWIW, this LGTM

**Pull request merged**